odoo/addons/base/ir/ir_mail_server.py

```diff
@@ -367,6 +367,7 @@ def build_email(self, email_from, email_to, subject, body, email_cc=None, email_
                 # so we fix it by using RFC2047 encoding for the filename instead.
                 part.set_param('name', filename_rfc2047)
                 part.add_header('Content-Disposition', 'attachment', filename=filename_rfc2047)
+                part.add_header('Content-ID', '<%s>' % fname)
```

Comment:
You can probably drop this as we need deeper changes to support inline images properly, especially if combined with other attachments. We would need to dynamically generate complex multipart structures:
```
  multipart/mixed
    multipart/related
      multipart/alternative
          text/plain: body with CIDs
          text/html: body with CIDs
      image/png <CID:1> (inline image)
   application/pdf (other attachment)
      

```